ci/Jenkinsfile-pyp

```
@@ -0,0 +1,205 @@
+            description: 'Indicates what kind of version bump should be done'
+        )
+        booleanParam(name: 'run_job', defaultValue: false, description: 'Allow for dry-runs')
+        booleanParam(name: 'upload', defaultValue: false, description: 'Publish to PyPi')
+    }
+
+    options {
+        disableConcurrentBuilds()
+        checkoutToSubdirectory('inmanta-sphinx')
+    }
+
+    stages {
+
+        stage("Setup venv and GitHub token") {
+            when {
+                expression { "${run_job}" == "true" }
+            }
+            steps {
+                dir('inmanta-sphinx') {
+                    sh '''
+                        python3 -m venv ${WORKSPACE}/env
+
+                        echo 'echo \${GITHUB_TOKEN}' > ${WORKSPACE}/ASK_PASS
+                        chmod u+x ${WORKSPACE}/ASK_PASS
+                    '''
+                }
+            }
+        }
+
+        stage("Setup next branch for major or minor release") {
+            when{
+                allOf {
+                    expression { "${run_job}" == "true" }
+                    expression { "${release_type}" == "stable" }
+                    anyOf {
+                        expression { "${version_increment}" == "major" }
+                        expression { "${version_increment}" == "minor" }
+                    }
+                }
+            }
+            steps{
+                dir('inmanta-sphinx') {
+                    withCredentials([string(credentialsId: 'fff7ef7e-cb20-4fb2-a93b-c5139463c6bf', variable: 'GITHUB_TOKEN')]) {
+                        sh '''
+                            export GIT_ASKPASS="${WORKSPACE}/ASK_PASS"
+                            ${WORKSPACE}/env/bin/python3 -m pip install -U pip bumpversion
+
+                            git checkout -b issue/version-bump
+
+                            # bump version if major release
+                            if [ "${version_increment}" == "major" ]; then
+                                ${WORKSPACE}/env/bin/python3 -m bumpversion --commit ${version_increment} CHANGELOG.md
+                            fi
+
+                            # set release date
+                            release_date=$(date --utc +'%y-%m-%d')
+                            sed -i "1s/\\?/${release_date}/" changelog.md
+                            git commit -m "set release date: ${release_date}" changelog.md
+
+                            git push origin issue/version-bump
+
+                            # Merge into next
+                            git checkout next
+                            git pull
+                            git merge -X theirs issue/version-bump
+                            git push origin next
+
+                            # Create a pull request to bump the version on master
+                            git checkout issue/version-bump
+                            ${WORKSPACE}/env/bin/python3 -m bumpversion "${version_increment}"
```

```suggestion
                            ${WORKSPACE}/env/bin/python3 -m bumpversion minor
```
The version of master should be the next minor release compared to the latest stable release.